### PR TITLE
Update TypeKind doc comment

### LINKED ISSUE #47946: TypeKind enum documentation does not indicate the value for records
https://github.com/dotnet/roslyn/blob/438047c55d476191fc5cb3df5f02925418139f1d/src/Compilers/Core/Portable/Symbols/TypeKind.cs#L9-L88

`TypeKind` doesn't contain a member called "Record". If this falls under TypeKind.Class, the documentation should reflect this. Otherwise, there should be a remarks section on `TypeKind` enum to clarify what's the case with records.

### PR DESCRIPTION
**Pull request by Youssef1313**
Fixes #47946

While a record is indeed a class, I think it's still beneficial to explicitly state that in the documentation comment.

### CODE REVIEW

**Comment by 333fred** [comment]
I don't think this is a good change. We're explicitly planning on record structs in C# 10, at which point this comment becomes incorrect.

**Comment by Youssef1313** [comment]
> I don't think this is a good change. We're explicitly planning on record structs in C# 10, at which point this comment becomes incorrect.

Should I reword it to something like `Type is a class (including record classes)`, or just close the PR and also the linked issue? Both options would make sense to me, so will let you decide.

**Comment by 333fred** [comment]
I'd be in favor of just closing, but I'll let others from @dotnet/roslyn-compiler chime in.

**Comment by jcouv** [comment]
I'd favor closing as well.
